### PR TITLE
Simplify lineage proof and remove chia-consensus singleton types

### PR DESCRIPTION
**Pull request by Rigidity**
There were duplicated types between `chia-wallet` and `chia-consensus`, so this makes `chia-consensus` use `chia-wallet`'s types instead and simplifies the API for lineage proofs.

### CODE REVIEW

File: crates/chia-consensus/src/fast_forward.rs [hunk 1]
```diff
@@ -1,43 +1,16 @@
 use crate::error::{Error, Result};
 use chia_protocol::Bytes32;
 use chia_protocol::Coin;
+use chia_wallet::singleton::SingletonArgs;
+use chia_wallet::singleton::SingletonSolution;
+use chia_wallet::singleton::SingletonStruct;
 use chia_wallet::singleton::SINGLETON_TOP_LAYER_PUZZLE_HASH;
```
Comment (arvidn) [inline crates/chia-consensus/src/fast_forward.rs:7]:
```suggestion
use chia_wallet::singleton::{SingletonArgs, SingletonSolution, SingletonStruct, SINGLETON_TOP_LAYER_PUZZLE_HASH};
```

File: crates/chia-consensus/src/fast_forward.rs [hunk 2]
```diff
@@ -1,43 +1,16 @@
...
-use clvm_traits::{FromClvm, ToClvm};
+use clvm_traits::FromClvm;
+use clvm_traits::ToClvm;
```
Comment (arvidn) [inline crates/chia-consensus/src/fast_forward.rs:9]: it's not clear that this is an improvement

File: crates/chia-wallet/src/proof.rs [hunk 3]
```diff
@@ -4,7 +4,7 @@ use clvm_traits::{FromClvm, ToClvm};
 #[derive(Debug, Clone, PartialEq, Eq, ToClvm, FromClvm)]
 #[cfg_attr(fuzzing, derive(arbitrary::Arbitrary))]
 #[clvm(untagged, tuple)]
-pub enum Proof {
+pub enum MaybeEveProof {
```
Comment (arvidn) [inline crates/chia-wallet/src/proof.rs:7]: this change seems unrelated. what prompted it?

File: crates/chia-wallet/src/puzzles/singleton.rs [hunk 4]
```diff
@@ -21,8 +21,8 @@ pub struct SingletonStruct {
 
 #[derive(Debug, Clone, PartialEq, Eq, ToClvm, FromClvm)]
 #[clvm(list)]
-pub struct SingletonSolution<I> {
-    pub proof: Proof,
+pub struct SingletonSolution<I, P = LineageProof> {
```
Comment (arvidn) [inline crates/chia-wallet/src/puzzles/singleton.rs:24]: doesn't the proof for a singleton always have to be a lineage proof?

File: crates/chia-consensus/src/fast_forward.rs (hunk 1 above)
Comment (Rigidity) [reply crates/chia-consensus/src/fast_forward.rs:7]: Done, it formats to multiple lines though

File: crates/chia-consensus/src/fast_forward.rs (hunk 2 above)
Comment (Rigidity) [reply crates/chia-consensus/src/fast_forward.rs:9]: Fixed

File: crates/chia-wallet/src/proof.rs (hunk 3 above)
Comment (Rigidity) [reply crates/chia-wallet/src/proof.rs:7]: Renamed for clarity, since most of the time it's going to be a LineageProof not an EveProof (only during launching is it Eve)

File: crates/chia-wallet/src/puzzles/singleton.rs (hunk 4 above)
Comment (Rigidity) [reply crates/chia-wallet/src/puzzles/singleton.rs:24]: When you mint an NFT, create a DID, or launch any other singleton, there's one coin where it's an EveProof

File: crates/chia-wallet/src/proof.rs (hunk 3 above)
Comment (Rigidity) [reply crates/chia-wallet/src/proof.rs:7]: I reverted the naming and use of generics, in favor of using match instead

**Review by arvidn — APPROVED** [review]

**Review by arvidn — APPROVED** [review]
